src/Avalonia.Base/AvaloniaObjectExtensions.cs

```diff
@@ -192,9 +192,9 @@ public static IObservable<T> GetObservable<T>(this IAvaloniaObject o, AvaloniaPr
         {
             return observable.Subscribe(e =>
             {
-                if (e.Sender is TTarget)
+                if (e.Sender is TTarget target)
```

Comment:
Maybe a dumb question but when is the target not the sender? Isn't the observable always the sender? Do we need to pass the target to the action?

Reply:
Sender in this case is the control that is raising the event. And there is a mismatch sometimes so `e.Sender` might not be `TTarget` especially for properties that are defined somewhere and then we use `AddOwner` to use it somewhere else. `AddOwner` does not require objects to be derived from each other so in some cases there is a mismatch. 

For example this happens to `Content` property that is defined on `ContentControl`, but we have `ContentPresenter` using it and subscribing via `AddClassHandler` so it ends up getting a `Window` sometimes, which in this case will end up with not being a `ContentPresenter` so the handler is not invoked.